sdk/azidentity/cache/darwin.go

```diff
@@ -0,0 +1,64 @@
+//go:build go1.18 && darwin
+// +build go1.18,darwin
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package cache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity/internal"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/log"
+	"github.com/AzureAD/microsoft-authentication-extensions-for-go/cache/accessor"
+	"github.com/AzureAD/microsoft-authentication-extensions-for-go/cache/accessor/file"
+)
+
+var cacheDir = os.UserHomeDir
+
+func storage(o internal.TokenCachePersistenceOptions) (accessor.Accessor, error) {
+	name := o.Name
+	if name == "" {
+		name = defaultName
+	}
+	var a accessor.Accessor
+	err := tryAccessor()
+	if err != nil {
+		msg := fmt.Sprintf("cache encryption is impossible because the keychain isn't usable: %s", err)
```

Comment:
What causes us to go down this path?

Reply:
Anything that prevents us from writing, reading, deleting a keychain item. I imagine the most common things will be running this code in an SSH session and the user not granting keychain access.